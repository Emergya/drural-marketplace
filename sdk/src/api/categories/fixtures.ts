export const categoryWithChildren = "Q2F0ZWdvcnk6Nw==";
export const categoryWithParent = "Q2F0ZWdvcnk6MjI=";
