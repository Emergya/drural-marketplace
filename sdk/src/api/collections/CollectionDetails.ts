import BaseDetails, { MapQueryData } from "../../helpers/BaseDetails";
import {
  CollectionDetails as CollectionDetailsQuery,
  CollectionDetailsVariables,
} from "../../queries/gqlTypes/CollectionDetails";
import { CollectionDetails as Collection } from "../../fragments/gqlTypes/CollectionDetails";
import { collectionDetails } from "../../queries/collections";

export class CollectionDetails extends BaseDetails<
  CollectionDetailsQuery,
  Collection,
  CollectionDetailsVariables
> {
  mapQueryData: MapQueryData<CollectionDetailsQuery, Collection> = data =>
    data.collection || undefined;

  query = (variables: CollectionDetailsVariables) =>
    this.client!.query<CollectionDetailsQuery, CollectionDetailsVariables>({
      fetchPolicy: "network-only",
      query: collectionDetails,
      variables,
    });
}
