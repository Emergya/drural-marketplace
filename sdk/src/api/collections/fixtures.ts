export const collectionId = "Q29sbGVjdGlvbjox";
export const collectionSlug = "summer-collection";
