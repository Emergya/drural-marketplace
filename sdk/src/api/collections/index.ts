export * from "./collections";
export * from "./CollectionList";
