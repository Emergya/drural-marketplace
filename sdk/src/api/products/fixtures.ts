export const productId = "UHJvZHVjdDo4NA==";
export const productSlug = "seaman-beer";
