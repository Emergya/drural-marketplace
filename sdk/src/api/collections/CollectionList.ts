import BaseList, {
  GetPageInfo,
  GetTotalCount,
  MapQueryData,
} from "../../helpers/BaseList";
import {
  CollectionList as CollectionListQuery,
  CollectionListVariables,
} from "../../queries/gqlTypes/CollectionList";
import { BaseCollection } from "../../fragments/gqlTypes/BaseCollection";
import { collections } from "../../queries/collections";

export class CollectionList extends BaseList<
  CollectionListQuery,
  BaseCollection,
  CollectionListVariables
> {
  getPageInfo: GetPageInfo<CollectionListQuery> = result =>
    result.data.collections?.pageInfo!;

  getTotalCount: GetTotalCount<CollectionListQuery> = result =>
    result.data.collections?.totalCount!;

  mapQueryData: MapQueryData<CollectionListQuery, BaseCollection> = data =>
    data.collections?.edges.map(({ node }) => node);

  query = (variables: CollectionListVariables) =>
    this.client!.query<CollectionListQuery, CollectionListVariables>({
      fetchPolicy: "network-only",
      query: collections,
      variables,
    });
}
