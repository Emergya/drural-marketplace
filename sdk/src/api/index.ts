import ApolloClient from "apollo-client";

import { LocalStorageManager } from "../data";
import { ApolloClientManager } from "../data/ApolloClientManager";
import { LocalStorageHandler } from "../helpers/LocalStorageHandler";
import { JobsManager } from "../jobs";
import { SaleorState } from "../state";
import { Config } from "../types";
import { AuthAPI } from "./Auth";
import APIProxy from "./APIProxy";
import { SaleorCartAPI } from "./Cart";
import { SaleorCheckoutAPI } from "./Checkout";
import { CollectionsAPI } from "./collections/collections";
import { CategoriesAPI } from "./categories/categories";
import { ProductsAPI } from "./products/products";

export * from "./Checkout";
export * from "./Cart";

export class SaleorAPI {
  auth: AuthAPI;

  checkout: SaleorCheckoutAPI;

  cart: SaleorCartAPI;

  categories: CategoriesAPI;

  collections: CollectionsAPI;

  products: ProductsAPI;

  /**
   * @deprecated Please do not use it anymore. Reference to API Proxy will be removed in future.
   * Now it just exists for legacy React hooks, which also will be removed.
   */
  legacyAPIProxy: APIProxy;

  constructor(
    client: ApolloClient<any>,
    apiProxy: APIProxy,
    config: Config,
    onStateUpdate?: () => any
  ) {
    this.legacyAPIProxy = apiProxy;

    const localStorageHandler = new LocalStorageHandler();
    const apolloClientManager = new ApolloClientManager(client);
    const jobsManager = new JobsManager(
      localStorageHandler,
      apolloClientManager
    );
    const saleorState = new SaleorState(
      config,
      localStorageHandler,
      apolloClientManager,
      jobsManager
    );
    const localStorageManager = new LocalStorageManager(
      localStorageHandler,
      saleorState
    );

    if (onStateUpdate) {
      saleorState.subscribeToNotifiedChanges(onStateUpdate);
    }

    this.auth = new AuthAPI(saleorState, jobsManager, config);
    this.checkout = new SaleorCheckoutAPI(saleorState, jobsManager, config);
    this.cart = new SaleorCartAPI(
      localStorageManager,
      apolloClientManager,
      saleorState,
      jobsManager,
      config
    );
    this.categories = new CategoriesAPI(client);
    this.collections = new CollectionsAPI(client, config);
    this.products = new ProductsAPI(client, config);

    this.legacyAPIProxy.attachAuthListener(authenticated => {
      if (!authenticated) {
        localStorageHandler.setCheckout({});
        localStorageHandler.setPayment({});
        localStorageHandler.setJobs(null);
      }
    });
  }
}
