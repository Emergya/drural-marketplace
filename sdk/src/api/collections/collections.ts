import ApolloClient from "apollo-client";
import {
  CollectionDetails as CollectionDetailsQuery,
  CollectionDetailsVariables,
} from "../../queries/gqlTypes/CollectionDetails";
import { CollectionDetails as CollectionDetailsFragment } from "../../fragments/gqlTypes/CollectionDetails";
import {
  CollectionList as CollectionListQuery,
  CollectionListVariables,
} from "../../queries/gqlTypes/CollectionList";
import { BaseCollection } from "../../fragments/gqlTypes/BaseCollection";
import { WithDetails, WithList } from "../types";
import { ConfigInput } from "../../types";
import { CollectionList } from "./CollectionList";
import { CollectionDetails } from "./CollectionDetails";

export class CollectionsAPI
  implements
    WithDetails<
      CollectionDetailsQuery,
      CollectionDetailsFragment,
      CollectionDetailsVariables
    >,
    WithList<CollectionListQuery, BaseCollection, CollectionListVariables> {
  private client: ApolloClient<any>;

  // temporary solution, might change in future
  private config: ConfigInput;

  constructor(client: ApolloClient<any>, config: ConfigInput) {
    this.client = client;
    this.config = config;
  }

  /**
   * Method returning collection details
   * @param variables Details parameters
   */
  getDetails = async (variables: CollectionDetailsVariables) => {
    const details = new CollectionDetails(this.client);

    await details.init({ channel: this.config.channel, ...variables });

    return details;
  };

  /**
   * Method returning list of collections with ability to request next page
   * @param params List parameters
   */
  getList = async (variables: CollectionListVariables) => {
    const list = new CollectionList(this.client);

    await list.init({ channel: this.config.channel, ...variables });

    return list;
  };
}
