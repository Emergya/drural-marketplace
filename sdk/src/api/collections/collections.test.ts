import ApolloClient from "apollo-client";
import { setupRecording, setupAPI } from "../../../testUtils/api";
import { CollectionsAPI } from "./collections";
import { defaultConfig } from "../../config";
import {
  OrderDirection,
  CollectionSortField,
} from "../../gqlTypes/globalTypes";
import * as fixtures from "./fixtures";

setupRecording();

describe("Collection object", () => {
  let client: ApolloClient<any>;
  let collectionsAPI: CollectionsAPI;

  beforeAll(async done => {
    client = (await setupAPI()).client;
    collectionsAPI = new CollectionsAPI(client, {
      ...defaultConfig,
      apiUrl: "",
      channel: "default-channel",
    });

    done();
  });

  it("can get a details of collection by id", async () => {
    const details = await collectionsAPI.getDetails({
      id: fixtures.collectionId,
    });

    expect(details.data).toMatchSnapshot();
    expect(details.loading).toBe(false);
  });

  it("can get a details of collection by slug", async () => {
    const details = await collectionsAPI.getDetails({
      slug: fixtures.collectionSlug,
    });

    expect(details.data).toMatchSnapshot();
    expect(details.loading).toBe(false);
  });

  it("can get a list of collections", async () => {
    const list = await collectionsAPI.getList({
      first: 20,
    });

    expect(list.data).toMatchSnapshot();
    expect(list.loading).toBe(false);
  });

  it("can get new page", async () => {
    const list = await collectionsAPI.getList({
      first: 1,
    });

    expect(list.data).toMatchSnapshot();
    expect(list.loading).toBe(false);

    list.next();

    expect(list.loading).toBe(true);

    await list.current;

    expect(list.data).toMatchSnapshot();
    expect(list.loading).toBe(false);
  });

  it("can sort", async () => {
    const list = await collectionsAPI.getList({
      first: 20,
      sortBy: {
        direction: OrderDirection.DESC,
        field: CollectionSortField.NAME,
      },
    });

    expect(list.data).toMatchSnapshot();
    expect(list.loading).toBe(false);
  });

  it("can filter", async () => {
    const list = await collectionsAPI.getList({
      filter: {
        search: "winter",
      },
      first: 20,
    });

    expect(list.data).toMatchSnapshot();
    expect(list.loading).toBe(false);
  });
});
