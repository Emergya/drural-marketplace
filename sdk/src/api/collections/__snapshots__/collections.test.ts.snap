// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection object can filter 1`] = `
Array [
  Object {
    "__typename": "Collection",
    "id": "Q29sbGVjdGlvbjoy",
    "name": "Winter sale",
    "seoDescription": null,
    "seoTitle": null,
    "slug": "winter-sale",
  },
]
`;

exports[`Collection object can get a details of collection by id 1`] = `
Object {
  "__typename": "Collection",
  "backgroundImage": Object {
    "__typename": "Image",
    "alt": "",
    "url": "https://master.staging.saleor.cloud/media/collection-backgrounds/summer.jpg",
  },
  "description": "{\\"blocks\\": [{\\"data\\": {\\"text\\": \\"The Saleor Summer Collection features a range of products that feel the heat of the market. A demo store for all seasons. Saleor captures the open source, e-commerce sun.\\"}, \\"type\\": \\"paragraph\\"}]}",
  "id": "Q29sbGVjdGlvbjox",
  "name": "Summer collection",
  "seoDescription": null,
  "seoTitle": null,
  "slug": "summer-collection",
}
`;

exports[`Collection object can get a details of collection by slug 1`] = `
Object {
  "__typename": "Collection",
  "backgroundImage": Object {
    "__typename": "Image",
    "alt": "",
    "url": "https://master.staging.saleor.cloud/media/collection-backgrounds/summer.jpg",
  },
  "description": "{\\"blocks\\": [{\\"data\\": {\\"text\\": \\"The Saleor Summer Collection features a range of products that feel the heat of the market. A demo store for all seasons. Saleor captures the open source, e-commerce sun.\\"}, \\"type\\": \\"paragraph\\"}]}",
  "id": "Q29sbGVjdGlvbjox",
  "name": "Summer collection",
  "seoDescription": null,
  "seoTitle": null,
  "slug": "summer-collection",
}
`;

exports[`Collection object can get a list of collections 1`] = `
Array [
  Object {
    "__typename": "Collection",
    "id": "Q29sbGVjdGlvbjoz",
    "name": "Featured Products",
    "seoDescription": null,
    "seoTitle": null,
    "slug": "featured-products",
  },
  Object {
    "__typename": "Collection",
    "id": "Q29sbGVjdGlvbjox",
    "name": "Summer collection",
    "seoDescription": null,
    "seoTitle": null,
    "slug": "summer-collection",
  },
  Object {
    "__typename": "Collection",
    "id": "Q29sbGVjdGlvbjoy",
    "name": "Winter sale",
    "seoDescription": null,
    "seoTitle": null,
    "slug": "winter-sale",
  },
]
`;

exports[`Collection object can get new page 1`] = `
Array [
  Object {
    "__typename": "Collection",
    "id": "Q29sbGVjdGlvbjoz",
    "name": "Featured Products",
    "seoDescription": null,
    "seoTitle": null,
    "slug": "featured-products",
  },
]
`;

exports[`Collection object can get new page 2`] = `
Array [
  Object {
    "__typename": "Collection",
    "id": "Q29sbGVjdGlvbjoz",
    "name": "Featured Products",
    "seoDescription": null,
    "seoTitle": null,
    "slug": "featured-products",
  },
  Object {
    "__typename": "Collection",
    "id": "Q29sbGVjdGlvbjox",
    "name": "Summer collection",
    "seoDescription": null,
    "seoTitle": null,
    "slug": "summer-collection",
  },
]
`;

exports[`Collection object can sort 1`] = `
Array [
  Object {
    "__typename": "Collection",
    "id": "Q29sbGVjdGlvbjoy",
    "name": "Winter sale",
    "seoDescription": null,
    "seoTitle": null,
    "slug": "winter-sale",
  },
  Object {
    "__typename": "Collection",
    "id": "Q29sbGVjdGlvbjox",
    "name": "Summer collection",
    "seoDescription": null,
    "seoTitle": null,
    "slug": "summer-collection",
  },
  Object {
    "__typename": "Collection",
    "id": "Q29sbGVjdGlvbjoz",
    "name": "Featured Products",
    "seoDescription": null,
    "seoTitle": null,
    "slug": "featured-products",
  },
]
`;
